fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/SplitHelper.java

```diff
@@ -827,13 +855,14 @@ private boolean appendFirst(@Nonnull KeyValue kv) {
             next = new KeyValue(nextPrefix, kv.getValue());
             nextIndex = keyTuple.getLong(keyTuple.size() - 1);
             sizeInfo.set(kv);
+            boolean done;
```

Comment:
Just a comment. The logic change looks ok but perhaps safer just add calls to the logging methods inline. I.e. if that else clause at the end didn't throw done would be unset.

Reply:
I felt like this was a better for maintainability than having the calls to the logger everywhere.